gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
+                GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
+                configuration.setGradleDistribution( from );
+                configuration.setProjectDir( projecLocation.toFile() );
+                configuration.setApplyWorkingSets( false );
+                configuration.setWorkingSets( new ArrayList<String>() );
+                SynchronizeGradleProjectJob synchronizeGradleProjectJob =
+                    new SynchronizeGradleProjectJob(
+                        configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
+                        AsyncHandler.NO_OP );
+
+                final IPath finalClassPath =
+                    getClassFilePath( projectName, className, packageName, projectTemplateName, projecLocation );
+
+                final File finalClassFile = finalClassPath.toFile();
+
+                if ( finalClassFile.exists() )
+                {
+                    NewLiferayModuleProjectOpMethods.addProperties( finalClassFile, properties );    
+                }
+
+                synchronizeGradleProjectJob.runInWorkspace( monitor );
+                synchronizeGradleProjectJob.setUser( true );
+                synchronizeGradleProjectJob.schedule();
+            }
+        }
+        catch( Exception e )
+        {
+            retval = ProjectCore.createErrorStatus( "can't create module project.", e );
         }
 
         return retval;
     }
 
+    private IPath getClassFilePath(
+        final String projectName, String className, final String packageName, final String projectTemplateName,
+        IPath projecLocation )
+    {
+        if( CoreUtil.isNullOrEmpty( className ) )
+        {
+            className = WordUtils.capitalize( projectName );
+        }
+
+        if( projectTemplateName.equals( "servicebuilder" ) || projectTemplateName.equals( "portlet" ) ||
+            projectTemplateName.equals( "mvcportlet" ) )
+        {
+            if( !className.contains( "Portlet" ) )
+            {
+                className += "Portlet";
+            }
+        }
+
+        final String finalClassName = className + ".java";
+
+        String[] finalPakcage = packageName.split( "\\." );
+
+        IPath packageNamePath = projecLocation.append( "src" ).append( "main" ).append( "java" );
+
+        for( String pack : finalPakcage )
+        {
+            packageNamePath = packageNamePath.append( pack );
+        }
+
+        final IPath finalClassPath = packageNamePath.append( finalClassName );
+
+        return finalClassPath;
+    }
+
+    private String checkForErrors( String[] lines )
```

Comment:
Add this method to BladeCLI.  the BladeCLI class should know how to parse the output of the blade command to search for "errors"